Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -3816,15 +3989,25 @@ namespace Web::Bindings {
 }
 )~~~");
 
-    // Generate an empty prototype object for global interfaces.
+    // Generate a mostly empty prototype object for global interfaces.
     auto is_global_interface = interface.extended_attributes.contains("Global");
     if (is_global_interface) {
         generator.append(R"~~~(
 void @prototype_class@::initialize(JS::Realm& realm)
 {
     Base::initialize(realm);
+)~~~");
+        if (interface.supports_named_properties()) {
```

Comment:
is there ever going to be another interface with [Global] ? Would be nice to clean this up and always do this for "any [Global] interface" rather than having a path for non-named-property-supporting global interfaces